pkg/ui/workspaces/crdb-js-proto/BUILD.bazel

```diff
@@ -0,0 +1,83 @@
+load("@aspect_rules_js//js:defs.bzl", "js_binary", "js_run_binary", "js_test")
+load("@aspect_rules_js//npm:defs.bzl", "npm_package")
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("//docs/generated/http:defs.bzl", "PROTOBUF_TARGETS")
+load("//pkg/ui/workspaces/db-console/src/js:defs.bzl", "proto_sources")
+
+npm_link_all_packages(name = "node_modules")
+
+genrule(
+    name = "protos",
+    outs = ["dist"],
+    cmd = """ \
+    mkdir -p $(OUTS); \
+    DESCRIPTOR_SET_IN=$$(for word in $(SRCS); do echo $$word; done | grep '\\.bin$$' | tr -s '\\n' ':'); \
+    PROTO_PATHS=$$(for f in $(locations :proto_sources); \
+        do if [[ "$$f" =~ ^.*/pkg/.*/_virtual_imports/.*_proto/.*.proto$$ ]]; \
+            then echo $$f | sed 's|.*pkg/.*/_virtual_imports/.*_proto/||' | sed 's|.bin$$||'; \
+        fi; done | sort -u); \
+    export PATH=$$PATH:$$(dirname $(NODE_PATH)); \
+    $(location @com_google_protobuf//:protoc) \
+    --plugin=$(location :node_modules/@bufbuild/protoc-gen-es/bin/protoc-gen-es) \
+    --es_out $(OUTS) \
+    --descriptor_set_in $$DESCRIPTOR_SET_IN \
+    --es_opt target=js+dts \
+    $$PROTO_PATHS \
+    """,
+    srcs = [
+        ":node_modules/@bufbuild/protoc-gen-es/bin/protoc-gen-es",
+        ":proto_sources",
```

Comment:
`:proto_sources` target allows us to obtain real file paths in project structure instead of defined paths in descriptor set (that can be different from actual .proto file path).
For instance, `pkg/build:build_proto` target includes `pkg/build/info.proto` file and it is named as `.../pkg/build/build_proto/build.proto.bin` which doesn't allow map this name to real path.

Instead, `:proto_sources` include real paths which easier to transform to relative path.